frontend/src/scenes/saved-insights/savedInsightsLogic.ts

```diff
@@ -124,8 +124,17 @@ export const savedInsightsLogic = kea<savedInsightsLogicType>({
                 return { ...values.insights, results: updatedInsights }
             },
             setInsight: (insight: InsightModel) => {
-                const results = values.insights.results.map((i) => (i.short_id === insight.short_id ? insight : i))
-                return { ...values.insights, results }
+                return {
+                    ...values.insights,
+                    results: values.insights.results.map((i) => (i.short_id === insight.short_id ? insight : i)),
+                }
+            },
+            addInsight: (insight: InsightModel) => {
```

Comment:
Have been looking a bit deeper into the kea code here after @macobo's comment. As I understand anything added to loaders would create related success and error actions etc. as well i.e. `addInsightSuccess` and `addInsightFailure` in this case. IMO it would be cleaner to have `setInsight` and `addInsight` in reducers, since it's a simple action -> state update. Don't know how we've handled this in the past and having the extra actions doesn't hurt as well.

Reply:
As far as I can tell we've handled it in both ways. E.g. `insightLogic` has both patterns, `savedInsightLogic` had `setInsight` already.

I'm torn... one way introduces actions we're not using, the other way means we need to look in two places in the file to understand what is happening. 

I guess [the four rules of simple design](https://martinfowler.com/bliki/BeckDesignRules.html) might help?

* Passes the tests -> yes
* Reveals intention -> being in a loader but not needing to load anything might not be intention revealing
* No duplication -> yes
* Fewest elements -> being in the loader adds more actions (even if we don't have to look at them)

So, I think I'd argue that in the reducers is better 👍